Blazor.Test/Shared/MainLayout.razor

```diff
@@ -0,0 +1,130 @@
+﻿@inherits LayoutComponentBase
+@inject IJSRuntime JS
+@inject NavigationManager Navigation
+
+    <CDTSLayout RefTop = "@refTopObj" Top = "@topObj" PreFooter = "@preFooterObj" Footer = "@footerObj" RefFooter = "@refFooterObj" LeftMenu = "@leftMenuObj">
+        <NavLink href="" Match="NavLinkMatch.All">
+            <button class="btn">Home</button>
+        </NavLink>
+        <NavLink href="externallink">
+            <button class="btn">External Link</button>
+        </NavLink>
+        <NavLink href="wetform">
+            <button class="btn">Wet Form</button>
+        </NavLink>
+            
+        <button class="btn" @onclick="AddSearch">Add Search</button>
+        <button class="btn" @onclick="AddFooter">Add Footer</button>
+        <button class="btn" @onclick="AddLeftMenu">Add left Menu</button>  
+        <button class="btn" @onclick="AddShareBtn">Add Share</button>  
+        @Body
+    </CDTSLayout>
+
+@code{
+
+    //Create your objects
+    GoC.WebTemplate.Blazor.Model.RefTop refTopObj = new GoC.WebTemplate.Blazor.Model.RefTop()
+    {
+        //For an application template, set IsApplication to true:
+        //IsApplication = true
```

Comment:
As discussed, this could be derived from the concrete type of the "Top" property.

Reply:
Done